services/minespace-web/src/HOC/AuthenticationGuard.js

```diff
@@ -24,77 +25,63 @@ import * as ENV from "@/constants/environment";
 const propTypes = {
   getUserInfoFromToken: PropTypes.func.isRequired,
   isAuthenticated: PropTypes.bool.isRequired,
-  keycloak: PropTypes.objectOf(PropTypes.any).isRequired,
+  authenticateUser: PropTypes.func.isRequired,
 };
 
 export const AuthenticationGuard = (isPublic) => (WrappedComponent) => {
   const authenticationGuard = (props) => {
-    const [authComplete, setAuthComplete] = useState();
     const { keycloak, initialized } = useKeycloak();
 
-    const authenticate = async () => {      
-      const authenticatingFromCoreFlag = localStorage.getItem("authenticatingFromCoreFlag");
-      const token = localStorage.getItem("jwt");
+    // get guid from pathname - props.location is not available at this level thus cannot directly access props.match.params.id
+    const guid = window.location.pathname.split("/mines/").pop().split("/")[0];
```

Comment:
We have started using `useParams` from `react-router-dom` in some places.  Is there a reason this wouldn't work here?